From 3f4d5ff34e061f3d47b9cf28bff2fc588d8c5326 Mon Sep 17 00:00:00 2001
From: psykose <alice@ayaya.dev>
Date: Thu, 23 Mar 2023 21:18:56 +0000
Subject: [PATCH] vulkan/wsi: use unsigned instead of uint
Upstream: https://gitlab.freedesktop.org/mesa/demos/-/commit/3f4d5ff34e061f3d47b9cf28bff2fc588d8c5326

uint is not a standard type name and fails to build against musl libc

Reviewed-by: Erik Faye-Lund <erik.faye-lund@collabora.com>
---
 src/vulkan/wsi/wayland.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/vulkan/wsi/wayland.c b/src/vulkan/wsi/wayland.c
index 444d2bb81..5958d283a 100644
--- a/src/vulkan/wsi/wayland.c
+++ b/src/vulkan/wsi/wayland.c
@@ -128,7 +128,7 @@ dispatch_key(xkb_keycode_t xkb_key, enum wl_keyboard_key_state state)
 }
 
 static void
-handle_key(uint key, enum wl_keyboard_key_state state)
+handle_key(unsigned key, enum wl_keyboard_key_state state)
 {
    xkb_keycode_t xkb_key = key + 8;
    struct itimerspec timer = {0};
@@ -151,15 +151,15 @@ handle_key(uint key, enum wl_keyboard_key_state state)
 }
 
 static void
-key(void *data, struct wl_keyboard *keyboard, uint serial,
-    uint time, uint key, enum wl_keyboard_key_state state)
+key(void *data, struct wl_keyboard *keyboard, unsigned serial,
+    unsigned time, unsigned key, enum wl_keyboard_key_state state)
 {
    handle_key(key, state);
 }
 
 static void
-modifiers(void *data, struct wl_keyboard *keyboard, uint serial,
-    uint mods_depressed, uint mods_latched, uint mods_locked, uint group)
+modifiers(void *data, struct wl_keyboard *keyboard, unsigned serial,
+    unsigned mods_depressed, unsigned mods_latched, unsigned mods_locked, unsigned group)
 {
    xkb_state_update_mask(keyboard_data.xkb_state, mods_depressed, mods_latched,
                          mods_locked, 0, 0, group);
@@ -184,7 +184,7 @@ keymap(void *data, struct wl_keyboard *keyboard,
 }
 
 static void
-enter(void *data, struct wl_keyboard *keyboard, uint serial,
+enter(void *data, struct wl_keyboard *keyboard, unsigned serial,
       struct wl_surface *surface, struct wl_array *keys)
 {
    uint32_t *key;
@@ -194,7 +194,7 @@ enter(void *data, struct wl_keyboard *keyboard, uint serial,
 }
 
 static void
-leave(void *data, struct wl_keyboard *keyboard, uint serial,
+leave(void *data, struct wl_keyboard *keyboard, unsigned serial,
       struct wl_surface *surface)
 {
    struct itimerspec timer = {0};
-- 
GitLab

